packages/splitter/src/useSplitter.ts

```diff
@@ -0,0 +1,274 @@
+    environment.removeEventListener('mousemove', onMouseMove);
+  });
+
+  const onTouchEnd = composeEventHandlers(props.onTouchEnd, () => {
+    // must remove global events on transaction finish
+    environment.removeEventListener('touchend', onTouchEnd);
+    environment.removeEventListener('touchmove', onTouchMove);
+  });
+
+  const onMouseDown = composeEventHandlers(props.onMouseDown, (event: React.MouseEvent) => {
+    event.preventDefault();
+    if (type === SplitterType.FIXED) {
+      if (dimension > min) {
+        setDimensionWithCache(min);
+      }
+      if (dimension < max) {
+        setDimensionWithCache(max);
+      }
+    } else {
+      // Must register global events to track mouse move outside the container
+      environment.addEventListener('mouseup', onMouseUp);
+      environment.addEventListener('mousemove', onMouseMove);
+    }
+  });
+
+  const onTouchStart = composeEventHandlers(props.onTouchStart, () => {
+    if (type === SplitterType.FIXED) {
+      if (dimension > min) {
+        setDimensionWithCache(min);
+      }
+      if (dimension < max) {
+        setDimensionWithCache(max);
+      }
+    } else {
+      // Must register global events to track mouse move outside the container
+      environment.addEventListener('touchend', onTouchEnd);
+      environment.addEventListener('touchmove', onTouchMove);
+    }
+  });
+
+  const toggleMinMax = useCallback(() => {
+    if (dimensionPreCacheRef.current === min) {
+      setDimensionWithCache(max);
+    } else {
+      setDimensionWithCache(min);
+    }
+  }, [dimensionPreCacheRef, setDimensionWithCache, min, max]);
+
+  const onKeyDown = composeEventHandlers(props.onKeyDown, (event: React.KeyboardEvent) => {
+    if (orientation === SplitterOrientation.VERTICAL) {
+      switch (event.key) {
+        case 'ArrowLeft':
+          type === SplitterType.VARIABLE && addDelta(-50);
+          break;
+        case 'ArrowRight':
+          type === SplitterType.VARIABLE && addDelta(50);
+          break;
+        case 'Enter':
+          toggleMinMax();
+          break;
+      }
+    } else {
+      switch (event.key) {
+        case 'ArrowUp':
+          type === SplitterType.VARIABLE && addDelta(-50);
+          break;
+        case 'ArrowDown':
+          type === SplitterType.VARIABLE && addDelta(50);
+          break;
+        case 'Enter': {
+          toggleMinMax();
+          break;
+        }
+      }
+    }
+  });
+
+  const getSeparatorProps = ({ ...other } = {}) => ({
+    'data-garden-container-id': 'containers.splitter',
+    'data-garden-container-version': PACKAGE_VERSION,
+    role: 'separator',
+    ref: separatorRef,
+    onMouseDown,
+    onKeyDown,
+    onTouchStart,
+    'aria-label': label,
+    'aria-labelledby': labelledBy,
+    'aria-controls': controls,
+    'aria-valuenow': dimension,
+    'aria-valuemin': min,
+    'aria-valuemax': max,
+    'aria-orientation': orientation,
+    tabIndex: 0,
+    ...other
+  });
```

Comment:
Do you think we should wrap `getSeparatorProps` function in a `useCallback` to prevent any dependents relying on this to re-render?